src/applications/claims-status/components/appeals-v2/AppealListItemV2.jsx

```diff
@@ -1,43 +1,78 @@
-import _ from 'lodash';
 import { Link } from 'react-router';
 import moment from 'moment';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import { getStatusContents } from '../../utils/appeals-v2-helpers';
+import {
+  APPEAL_TYPES,
+  EVENT_TYPES,
+  getTypeName,
+  getStatusContents,
+} from '../../utils/appeals-v2-helpers';
 
-// TODO: Get a proper mapping of programArea -> display output
 const appealTypeMap = {
-  compensation: 'Compensation',
+  compensation: 'Disability Compensation',
   pension: 'Pension',
   insurance: 'Insurance',
   loan_guaranty: 'Loan Guaranty', // eslint-disable-line
   education: 'Education',
-  vre: 'VRE',
-  medical: 'Medical',
-  burial: 'Burial',
-  bva: 'BVA',
-  other: 'Other',
-  multiple: 'Multiple',
+  vre: 'Vocational Rehabilitation and Employment',
+  medical: 'Medical Benefits',
+  burial: 'Burial Benefits',
+  fiduciary: 'Fiduciary',
 };
 
 export default function AppealListItem({ appeal, name }) {
   const { status } = appeal.attributes;
-  // always show merged event on top
-  const events = _.orderBy(
-    appeal.attributes.events,
-    [e => e.type === 'merged', e => moment(e.date).unix()],
-    ['desc', 'desc'],
+
+  let requestEventType;
+  switch (appeal.type) {
+    case APPEAL_TYPES.legacy:
+      requestEventType = EVENT_TYPES.nod;
+      break;
+    case APPEAL_TYPES.supplementalClaim:
+      requestEventType = EVENT_TYPES.scRequest;
+      break;
+    case APPEAL_TYPES.higherLevelReview:
+      requestEventType = EVENT_TYPES.hlrRequest;
+      break;
+    case APPEAL_TYPES.appeal:
+      requestEventType = EVENT_TYPES.amaNod;
+      break;
+    default:
+    // do nothing
+  }
+
+  const requestEvent = appeal.attributes.events.find(
+    a => a.type === requestEventType,
   );
-  const firstEvent = events[events.length - 1];
+  const appealType = appealTypeMap[appeal.attributes.programArea];
+  const isAppeal =
+    appeal.type === APPEAL_TYPES.appeal || appeal.type === APPEAL_TYPES.legacy;
+
+  let appealTitle = '';
```

Comment:
suggest using template string instead of string concatenation here. maybe something like:
```
let appealTitle = isAppeal ? 
`${appealType || ''} ${getTypeName(appeal)} ${requestDisplayDate || ''}` : 
`${getTypeName(appeal)} ${appealType || ''} ${requestDisplayDate || ''}`;
appealTitle.trim();
```

```

Reply:
adding a comment for each template with an example of the full output would be useful too

Reply:
I tried out the template string approach. Your suggestion omits the words "for" and "Received", which are conditional on the presence of appealType and requestDisplayDate, respectively. It gets messy fast. I agree that my version is not the greatest. The comment with example output will help, though.